app/lib/missing_permission.rb

```diff
@@ -0,0 +1,9 @@
+class MissingPermission < RuntimeError
```

Comment:
Could this be namespaced in the `ProviderInterface`?

Reply:
Yep but not in `app/lib`.